core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -507,15 +508,17 @@ class ReplicaManager(val config: KafkaConfig,
    */
   def readFromLocalLog(fetchOnlyFromLeader: Boolean,
                        readOnlyCommitted: Boolean,
+                       fetchMaxBytes: Int,
                        readPartitionInfo: Map[TopicAndPartition, PartitionFetchInfo]): Map[TopicAndPartition, LogReadResult] = {
 
+    var limitBytes = fetchMaxBytes
     readPartitionInfo.map { case (TopicAndPartition(topic, partition), PartitionFetchInfo(offset, fetchSize)) =>
       BrokerTopicStats.getBrokerTopicStats(topic).totalFetchRequestRate.mark()
       BrokerTopicStats.getBrokerAllTopicsStats().totalFetchRequestRate.mark()
 
       val partitionDataAndOffsetInfo =
         try {
-          trace("Fetching log segment for topic %s, partition %d, offset %d, size %d".format(topic, partition, offset, fetchSize))
+          trace("Fetching log segment for topic %s, partition %d, offset %d, size %d, limit %d".format(topic, partition, offset, fetchSize, limitBytes))
```

Comment:
It's not very clear what's the difference btw size and limit from the log message. Could we change it to partition size and response limit?